jwst/barshadow/barshadow_step.py

```diff
@@ -0,0 +1,50 @@
+from ..stpipe import Step
+from .. import datamodels
+from . import bar_shadow
+
+
+class BarShadowStep(Step):
+    """
+    BarShadowStep: Inserts the bar shadow and wavelength arrays
+    into the data.
+
+    Bar shadow correction depends on the position of a pixel along the slit
+    and the wavelength. It is only applied to uniform sources and only for
+    NRS MSA data. 
+
+    """
+
+    spec = """
+    """
+
+    reference_file_types = ['barshadow']
+
+    def process(self, input):
+
+        # Open the input data model
+        with datamodels.open(input) as input_model:
+
+            # Get the name of the bar shadow reference file to use
```

Comment:
If this only gets applied to NRS_MSASPEC mode, should we have a check up here at the very front that skips the step if the data are not from that mode? When run from the calspec2 pipeline the pipeline module can have the logic to only call the step for that exp_type, but we should also have a check in the step to prevent a user from applying just that step to non-MSA exposures.

Reply:
Is there a `meta.cal_step.barshadow` keyword that can be set to "SKIP" or "COMPLETE"? in the step?